src/dune_action/examples/race2.ml

```diff
@@ -0,0 +1,20 @@
+open Dune_action
+
+let contains equal list elem = List.find list (equal elem) |> Option.is_some
+
+(* More subtle race. Should we allow target that is inside a directory that is
```

Comment:
Targets are only allowed in the directory the rule is defined in, so no? For symlinks, I think dune expects the actions to follow symlinks and targets should never be symlinks. (I might be wrong about that)